Mathlib/FieldTheory/SeparableDegree.lean

```diff
@@ -476,14 +472,18 @@ theorem natSepDegree_expand_eq_natSepDegree (q : ℕ) [hF : ExpChar F q] {n : 
   cases' hF with _ _ hprime _
   · simp only [one_pow, expand_one]
   haveI := Fact.mk hprime
-  simpa only [natSepDegree_eq_natSepDegree_algebraicClosure, aroots_def, map_expand,
+  simpa only [natSepDegree_eq_of_isAlgClosed (AlgebraicClosure F), aroots_def, map_expand,
     Fintype.card_coe] using Fintype.card_eq.2 ⟨rootsExpandPowEquivRoots q
       (f := f.map (algebraMap F (AlgebraicClosure F))) (n := n)⟩
 
+namespace HasSeparableContraction
+
+variable {f}
+
 /-- If a polynomial has separable contraction, then its separable degree is equal to its
 separable contraction degree. -/
-theorem natSepDegree_eq_hasSeparableContraction_degree
-    (q : ℕ) [ExpChar F q] (hf : f.HasSeparableContraction q) :
+theorem natSepDegree_eq
+    {q : ℕ} [ExpChar F q] (hf : f.HasSeparableContraction q) :
```

Comment:
You could do
```suggestion
variable {f} in
theorem HasSeparableContraction.natSepDegree_eq
    {q : ℕ} [ExpChar F q] (hf : f.HasSeparableContraction q) :
```
without adding a `namespace` section.

Reply:
Yes, but `variable {f} in` needs to be written before the comments; I don't know why.

Reply:
I think `in` is a macro that gets expanded to
```
section
variable {f}
...
end
```
so that explains it.